plugins/utils/plugin/commandutils.go

```diff
@@ -0,0 +1,83 @@
+// Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package plugin
+
+import (
+	log "github.com/cihub/seelog"
+	"github.com/coreos/go-iptables/iptables"
+)
+
+// RedirectNonLocalTraffic adds iptable rules to the given chain to route non-local traffic coming in from the
+// given listener port to redirect port and adds the chain to the NAT table at PREROUTING stage
+func RedirectNonLocalTraffic(
+	iptable *iptables.IPTables,
+	chain string,
+	redirectPortToListenerPorts map[string]string) error {
+
+	// Create a new chain
+	err := iptable.NewChain("nat", chain)
+	if err != nil {
+		return err
+	}
+
+	// Route everything arriving at listener ports to intercept port.
+	for redirectPort, listenerPorts := range redirectPortToListenerPorts {
+		err := iptable.Append("nat", chain, "-p", "tcp", "-m", "multiport", "--dports", listenerPorts,
+			"-j", "REDIRECT", "--to-port", redirectPort)
```

Comment:
I guess this comes from appmesh? But yes, this is confusing now (maybe even invalid, depending on the actual values passed)

Reply:
Yes thats right. we are doing the same here.

Reply:
I think this still looks backwards (or at least the terminology is wrong)

`--dports` is the port to be intercepted. In case of SC, this is `interceptPort`, say `8080`.
`--to-port` is the final destination for the traffic. In case of SC, this should be port envoy listens to.

Reply:
The values are correct. I think this is confusing because of the names. Renaming it.